core/src/main/java/org/elasticsearch/common/transport/BoundTransportAddress.java

```diff
@@ -62,18 +65,38 @@ public static BoundTransportAddress readBoundTransportAddress(StreamInput in) th
 
     @Override
     public void readFrom(StreamInput in) throws IOException {
-        boundAddress = TransportAddressSerializers.addressFromStream(in);
+        int boundAddressLength = in.readInt();
+        boundAddresses = new TransportAddress[boundAddressLength];
+        for (int i = 0; i < boundAddressLength; i++) {
+            boundAddresses[i] = TransportAddressSerializers.addressFromStream(in);
+        }
         publishAddress = TransportAddressSerializers.addressFromStream(in);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        TransportAddressSerializers.addressToStream(out, boundAddress);
+        out.writeInt(boundAddresses.length);
+        for (TransportAddress address : boundAddresses) {
+            TransportAddressSerializers.addressToStream(out, address);
+        }
         TransportAddressSerializers.addressToStream(out, publishAddress);
     }
 
     @Override
     public String toString() {
```

Comment:
Yeah this is what gets printed out. This is the current output:

```
[2015-09-21 13:42:27,541][INFO ][org.elasticsearch.http.netty] [Pyro] Bound http to address {127.0.0.1:9200}
[2015-09-21 13:42:27,542][INFO ][org.elasticsearch.http.netty] [Pyro] Bound http to address {[fe80::1]:9200}
[2015-09-21 13:42:27,543][INFO ][org.elasticsearch.http.netty] [Pyro] Bound http to address {[::1]:9200}
[2015-09-21 13:42:27,544][INFO ][org.elasticsearch.http   ] [Pyro] publish_address {127.0.0.1:9200}, bound_addresses {127.0.0.1:9200}, {[fe80::1]:9200}, {[::1]:9200}
```

I'll make the INFO level per address messages DEBUG.